src/dashboard/events/cd-dashboard-upcoming-event.vue

```diff
@@ -0,0 +1,216 @@
+<template>
+  <div class="cd-dashboard-upcoming-event">
+    <div v-if="event && event.id && dojo && dojo.id && orders !== null" class="cd-dashboard-upcoming-event__content" :class="{ 'cd-dashboard-upcoming-event__content--booked': hasOrder}">
+      <div class="cd-dashboard-upcoming-event__main">
+        <h4 v-if="hasOrder">{{ $t('Your next event is "{name}"', { name: event.name }) }}</h4>
+        <h4 v-else>{{ $t('"{name}" is the next session available', { name: event.name }) }}</h4>
+        <router-link v-if="!isChampion && !hasOrder && remainingTickets > 0" class="cd-dashboard-upcoming-event__book" :to="{ name: 'EventSessions', params: { eventId: event.id } }">{{ $t('Book now') }}</router-link>
+        <router-link v-if="hasOrder"  class="cd-dashboard-upcoming-event__link" :to="{ name: 'EventSessions', params: { eventId: event.id } }">
+          <span v-if="ninjaTickets > 0">{{ $t('{num} "{type}" tickets booked', { num: ninjaTickets, type: $t('Youth') }) }}</span>
+          <span v-if="mentorTickets > 0">{{ $t('{num} "{type}" tickets booked', { num: mentorTickets, type: $t('Mentor') }) }}</span>
+        </router-link>
+        <a v-if="isChampion" class="cd-dashboard-upcoming-event__link" href="#">
+          <span>{{ $t('{booked}/{total} {type} booked', { booked: bookedNinjaTickets, total: totalNinjaTickets, type: 'Youth' }) }}</span>
+          <span>{{ $t('{booked}/{total} {type} booked', { booked: bookedMentorTickets, total: totalMentorTickets, type: 'Mentor' }) }}</span>
+        </a>
+      </div>
+      <div class="cd-dashboard-upcoming-event__dojo">
+          <h4>{{ dojo.name }}</h4>
+          <p>{{ eventDate | cdDateFormatter }}, {{ formattedStartTime }} - {{ formattedEndTime }}</p>
+      </div>
+    </div>
+    <div v-else class="cd-dashboard-upcoming-event__filler cd-filler"></div>
+  </div>
+</template>
+
+<script>
+  import { mapGetters } from 'vuex';
+  import ImgFallback from '@/common/directives/cd-img-fallback';
+  import cdDateFormatter from '@/common/filters/cd-date-formatter';
+  import cdTimeFormatter from '@/common/filters/cd-time-formatter';
+  import EventsService from '@/events/service';
+  import EventUtils from '@/events/util';
+
+  export default {
+    name: 'cd-dashboard-upcoming-event',
+    props: ['event', 'dojo'],
+    directives: {
+      ImgFallback,
+    },
+    filters: {
+      cdDateFormatter,
+      cdTimeFormatter,
+    },
+    data() {
+      return {
+        orders: null,
+      };
+    },
+    computed: {
+      ...mapGetters(['loggedInUser']),
+      eventDate() {
+        return EventUtils.getNextStartTime(this.event);
+      },
+      remainingTickets() {
+        return this.totalNinjaTickets - this.bookedNinjaTickets;
+      },
+      bookedNinjaTickets() {
+        return this.event.sessions.reduce((acc1, session) =>
+          acc1 + session.tickets.filter(ticket => ticket.type === 'ninja').reduce((acc2, ticket) =>
```

Comment:
I think both approvedApplications && quantity deserve a function that is being called to calculate that taking `userType` as a param, and eventually `target` as approvedApplications or quantity.